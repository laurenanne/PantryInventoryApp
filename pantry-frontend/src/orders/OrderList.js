import React, { useState, useEffect } from "react";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import PantryApi from "../pantryApi";
import CircularProgress from "@mui/material/CircularProgress";
import { Typography } from "@mui/material";
import Box from "@mui/material/Box";
import OrderCard from "./OrderCard";

// Displays a page with list of all orders
// On mount loads from API
// Routed as "/orders"

// Routes to -> OrderCard

function OrderList() {
  const [orders, setOrders] = useState([]);

  // upon loading, call API to get a list of all food
  useEffect(() => {
    async function getOrders() {
      let orders = await PantryApi.getAllOrders();
      setOrders(orders);
    }
    getOrders();
  }, []);

  if (!orders) {
    return (
      <div>
        <CircularProgress color="secondary" />;
      </div>
    );
  } else {
    return (
      <React.Fragment>
        <Box sx={{ mr: 3, ml: 3 }}>
          <Typography sx={{ mb: 1, fontSize: "2rem", ml: 2, mt: 2 }}>
            Client Orders
          </Typography>
          <Table size="small">
            <TableHead>
              <TableRow>
                <TableCell>Order Id</TableCell>

                <TableCell>Date</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {orders.map((o) => (
                <OrderCard
                  key={o.orderId}
                  orderId={o.orderId}
                  clientId={o.clientId}
                  date={o.date}
                />
              ))}
            </TableBody>
          </Table>
        </Box>
      </React.Fragment>
    );
  }
}

export default OrderList;
