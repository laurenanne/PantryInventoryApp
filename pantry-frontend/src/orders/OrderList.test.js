import React from "react";
import { render } from "@testing-library/react";
import OrderList from "./OrderList";

it("matches snapshot", function() {
  const { asFragment } = render(<OrderList />);
  expect(asFragment()).toMatchSnapshot();
});
