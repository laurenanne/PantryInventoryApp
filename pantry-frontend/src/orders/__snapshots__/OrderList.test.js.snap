// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-rr8p7q"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-xedo3v-MuiTypography-root"
    >
      Client Orders
    </p>
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            Order Id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            Date
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      />
    </table>
  </div>
</DocumentFragment>
`;
